libs/librepcb/projecteditor/boardeditor/fsm/bes_addvia.cpp

```diff
@@ -368,31 +485,116 @@ void BES_AddVia::sizeEditValueChanged(const PositiveLength& value) noexcept {
+      noexcept {
   mCurrentViaDrillDiameter = value;
   if (mViaEditCmd) {
     mViaEditCmd->setDrillDiameter(mCurrentViaDrillDiameter, true);
   }
 }
 
 void BES_AddVia::setNetSignal(NetSignal* netsignal) noexcept {
+  Q_ASSERT(mSubState == SubState_PositioningVia);
   try {
-    if (!netsignal) throw LogicError(__FILE__, __LINE__);
-    mCurrentViaNetSignal = netsignal;
+    if (!netsignal) {
+      throw LogicError(__FILE__, __LINE__);
+    }
     mUndoStack.appendToCmdGroup(
         new CmdBoardNetSegmentRemove(mCurrentVia->getNetSegment()));
-    CmdBoardNetSegmentEdit* cmdEdit =
-        new CmdBoardNetSegmentEdit(mCurrentVia->getNetSegment());
-    cmdEdit->setNetSignal(*mCurrentViaNetSignal);
-    mUndoStack.appendToCmdGroup(cmdEdit);
+    QScopedPointer<CmdBoardNetSegmentEdit> cmdEdit(
+        new CmdBoardNetSegmentEdit(mCurrentVia->getNetSegment()));
+    cmdEdit->setNetSignal(*netsignal);
+    mUndoStack.appendToCmdGroup(cmdEdit.take());
     mUndoStack.appendToCmdGroup(
         new CmdBoardNetSegmentAdd(mCurrentVia->getNetSegment()));
   } catch (const Exception& e) {
     QMessageBox::critical(&mEditor, tr("Error"), e.getMsg());
   }
 }
 
+QSet<NetSignal*> BES_AddVia::getNetSignalsAtScenePos(Board& board,
+                      const Point& pos, QSet<BI_Base*> except) const noexcept {
+  QSet<NetSignal*> result = QSet<NetSignal*>();
+  foreach(BI_Via* via, board.getViasAtScenePos(pos)) {
+    if (except.contains(via)) continue;
+    if (!result.contains(&via->getNetSignalOfNetSegment())) {
+      result.insert(&via->getNetSignalOfNetSegment());
+    }
+  }
+  foreach(BI_NetPoint* netpoint, board.getNetPointsAtScenePos(pos)) {
+    if (except.contains(netpoint)) continue;
+    if (!result.contains(&netpoint->getNetSignalOfNetSegment())) {
+      result.insert(&netpoint->getNetSignalOfNetSegment());
+    }
+  }
+  foreach(BI_NetLine* netline, board.getNetLinesAtScenePos(pos)) {
+    if (except.contains(netline)) continue;
+    if (!result.contains(&netline->getNetSignalOfNetSegment())) {
+      result.insert(&netline->getNetSignalOfNetSegment());
+    }
+  }
+  foreach(BI_FootprintPad* pad, board.getPadsAtScenePos(pos)) {
+    if (except.contains(pad)) continue;
+    if (!result.contains(pad->getCompSigInstNetSignal())) {
+      result.insert(pad->getCompSigInstNetSignal());
+    }
+  }
+  return result;
+}
+
+NetSignal* BES_AddVia::getClosestNetSignal(Board& board, const Point& pos)
+noexcept {
+  //TODO(5n8ke): Get the closest candidate, instead of the most used
+  // for now a _closest_ NetSignal is only found, when it is at pos.
+  // Otherwise the last candidate is returned.
+  if (mFindClosestNetSignal) {
+    if (BI_NetLine* atPosition = findNetLine(board, pos)) {
+      mLastClosestNetSignal = &atPosition->getNetSignalOfNetSegment();
+    }
+    mFindClosestNetSignal = false;
+    QTimer *timer = new QTimer(this);
+    connect(timer, &QTimer::timeout, [this]() {
+      this->mFindClosestNetSignal = true;
+    });
```

Comment:
This regularly instantiates new `QTimer` objects which will be deleted only when closing the board editor. So the memory consumption constantly increases. I'd suggest to either re-use the same timer instance (by storing it as a member variable), or delete it after it expired:

```suggestion
    QTimer *timer = new QTimer(this);
    connect(timer, &QTimer::timeout, [this, timer]() {
      this->mFindClosestNetSignal = true;
      timer->deleteLater();
    });
```